crates/core/tedge_watchdog/src/systemd_watchdog.rs

```diff
@@ -113,6 +119,31 @@ async fn monitor_tedge_service(
     }
 }
 
+async fn get_latest_health_status_message(
+    request_timestamp: i64,
+    messages: &mut mpsc::UnboundedReceiver<Message>,
```

Comment:
```suggestion
messages: &mut (impl StreamExt<Item = Message>),
```

Reply:
Couldn't be done due the following limitation: `the trait Unpin is not implemented for impl StreamExt<Item = Message>`